### PR TITLE
feat(eval): GR00T policy eval recipe for the Isaac Lab runner (odyssey#17)

### PR DESCRIPTION
**Pull request by Danny024**
## What
Fills the `eval_script` slot the subprocess `IsaacLabRunner` leaves open: the **blessed GR00T evaluation recipe**. The runner owns launch / the `ODYSSEY_*` stdout protocol / cancellation / scoring; this recipe owns the GR00T↔Isaac obs/action loop — boots the Franka visuomotor env, drives a GR00T policy server, and reports each episode.

## Changes
- `scripts/eval/gr00t_isaac_eval.py` — the recipe. Accepts the launch contract (`--task/--num_episodes/--checkpoint/--headless` + config passthrough), connects to `run_gr00t_server`, replays GR00T action chunks in the IK-rel env, emits `ODYSSEY_EPISODE`/`ODYSSEY_RESULT`. Heavy deps (isaaclab/gymnasium/gr00t/torch/numpy) are deferred into the run path so the argv + protocol surface unit-test on a CPU box without Isaac. Optional `--video_dir` captures Isaac's viewport render (3rd-person, 720p).
- `scripts/eval/gr00t_transforms.py` — pure-numpy obs/action transforms (rot6d/quat/axis-angle; nested obs build; action decode).
- `tests/unit/test_gr00t_isaac_eval.py` — parser + `ODYSSEY_*` lines verified through the runner's own `EvalProtocolCollector` + `summarize`.
- `tests/unit/test_gr00t_transforms.py` — transform conventions (numpy-gated).
- `examples/quickstart-gr00t/isaac_eval_mission.yaml` — runnable mission wiring the recipe.

## Validation
Ran live on an NVIDIA H100 (GR00T server on GPU + Isaac Lab cosmos env): episode completed, `success_rate 0.0` (expected zero-shot for base GR00T) — the GR00T-in-Isaac loop is proven. Also driven end-to-end from Command Center via the CC→Odyssey delegation bridge.

### CODE REVIEW

**Comment by Danny024** [comment]
Wired the quickstart's `eval-in-isaac-lab` task to this recipe + merged #18's train fix, so the quickstart mission now runs **train → deploy-in-Isaac in a single `odyssey run`** (no more placeholder eval).

**Verified end-to-end on an H100:**
```
MISSION: COMPLETED  (overall_grade 0.000)
  finetune-gr00t     → Training completed (GR00T-N1.7-3B on cube_to_bowl_5, loss 1.06)
  eval-in-isaac-lab  → task.completed   success_rate 0.0
       ODYSSEY_EPISODE 1/2 + 2/2  (GR00T deployed in the cosmos cube-stack env)
```

Changes: eval benchmark → `Isaac-Stack-Cube-Franka-IK-Rel-Visuomotor-Cosmos-v0` (the recipe needs camera obs), `config.eval_script` → `scripts/eval/gr00t_isaac_eval.py` + host/port/instruction/pos_scale. Spec test updated. 220 unit tests green.

**Honest note:** the eval deploys **base GR00T (zero-shot)**, not the just-finetuned checkpoint — the finetune uses the `new_embodiment`/SO100 head while the cosmos eval is the DROID embodiment. A *matched* train→deploy of the same checkpoint needs aligned train/eval embodiment+task (the LIBERO route) — tracked as a follow-up.
